### ISSUE TITLE
Test Explorer will not show our tests unless the test project stops targeting net35

### ISSUE THREAD

**Issue by jnm2** [nunit/nunit#3157]
Affects Visual Studio 2017 Update 9 and Visual Studio 2019 previews.

If `net35` is present at all, even if it's not the first item, no tests show up in Test Explorer.

https://github.com/nunit/nunit/blob/b5c1d549938476ff06f118557b29778b36593549/src/NUnitFramework/tests/nunit.framework.tests.csproj#L4

https://github.com/nunit/nunit/blob/b5c1d549938476ff06f118557b29778b36593549/src/NUnitFramework/nunitlite.tests/nunitlite.tests.csproj#L4

The Tests output is this:

> Microsoft.VisualStudio.TestPlatform.ObjectModel.TestPlatformException: Framework35 is not supported. For projects targeting .Net Framework 3.5, please use Framework40 to run tests in CLR 4.0 "compatibility mode".

```
   at Microsoft.VisualStudio.TestPlatform.CommandLine.TestPlatformHelpers.TestRequestManager.UpdateRunSettingsIfRequired(String runsettingsXml, List`1 sources, String& updatedRunSettingsXml)
   at Microsoft.VisualStudio.TestPlatform.CommandLine.TestPlatformHelpers.TestRequestManager.RunTests(TestRunRequestPayload testRunRequestPayload, ITestHostLauncher testHostLauncher, ITestRunEventsRegistrar testRunEventsRegistrar, ProtocolConfig protocolConfig)
   at Microsoft.VisualStudio.TestPlatform.Client.DesignMode.DesignModeClient.<>c__DisplayClass19_0.<StartTestRun>b__0()
```

Since Test Explorer is the only remotely convenient way to debug a single test (unless I install ReSharper into this instance of VS), this is extremely frustrating.

I don't think there's anything we can do about it besides report it to Microsoft. Visual Studio should be applying `Framework40` itself; it's not us who are passing `Framework35` to VSTest.

@OsirisTerje, is this something we should tag a VSTest team member for?

**Comment by jnm2** [comment]
@OsirisTerje ping.

**Comment by icnocop** [comment]
Is https://github.com/Microsoft/vstest/issues/1896 related?

**Comment by mikkelbu** [comment]
@jnm2 Accordingly to Microsoft/vstest#1896: "This issue has been resolved in VS 2017 v15.9.10 (#1945).", and it again works on my machine.

**Comment by jnm2** [comment]
@icnocop @mikkelbu That's great news! Thank you!

(@mikkelbu I edited the link to point to their `#1945`. Hope that's okay.)
